torch/fx/passes/shape_prop.py

```diff
@@ -28,7 +29,50 @@ class TensorMetadata(NamedTuple):
     is_quantized : bool
     qparams: Dict[str, Any]
 
-def _extract_tensor_metadata(result : torch.Tensor, include_contiguity=True) -> TensorMetadata:
+    # Sparse metadata when layout in [ torch.sparse_coo/csr/csc/bsr/bsc ]
+    #   batch_dim + sparse_dim + dense_dim = ndim = len(shape)
+    #   blocksize for bsr/bsc
+    #   idx_dtype denote the types used for compressed indices and positions
+    layout : torch.layout
+    batch_dim : int
```

Comment:
sparse coo tensors don't have batch dimensions, so it could be Optional.

Reply:
happy to do that, but wouldn't batch_dim = 0 make sense here too? perhaps in the future we want to support batched COO (sparse JAX does that for example)?

Reply:
yes, `batch_dim = 0` makes sense for sparse_coo and even for strided layout. However, introducing batch dimensions to these layouts is OT for this PR.

In my experience, PyTorch PRs must follow the principle of minimality and necessity, that is, one should focus only at the feature at hand and consider everything else as constant (except for fixing trivial/obvious bugs).

[OT] Possible extensions should be discussed separately and when there is a consesus for implementing the extension, the corresponding PRs will address all required changes including new implementations, adjusting existing code, updating API and docs, introducing tests, etc. In long run, this makes a lot of sense as the code will not be flooded with incomplete/untested/broken features that may never be fixed/implemented because of changing priorities, for instance.